### PR TITLE
Feature/init routing

### PR DESCRIPTION
**Pull request by JoeyStewart**
What I did:

I added routing for the main page and for any article that is searched for. Some minor changes to the fetch calls were made. I did copy and paste the entirety of the original code at the bottom of the file for future reference.

Did this break anything?

- [X] No
- [ ]  Yes
Type of change

- [X] New feature (non-breaking change which adds functionality)
- [ ]  Styling 
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [X]  If this code needs to be tested, all tests are passing
- [X ]  The code runs locally
- [X ]  There are comments where clarification/ organization is needed
- [X ]  The code is DRY. If it's not, I tried my best
- [X ]  I reviewed my code before pushing

What's next?

### CODE REVIEW

**Review by tenthwalker — APPROVED** [review]
Great use of built-in hooks!

File: src/App/App.js [hunk 1]
```diff
@@ -9,9 +10,13 @@ function App() {
         <h1 className='header-text'>H8rAid!</h1>
         <button className='login-button'>Login</button>
       </header>
-        <WikipediaSearch />
+      <Routes>
+        <Route path='/' element={<WikipediaSearch />} />
+        <Route path='/main' element={<Navigate to='/' />} />
+        <Route path="article/:id" element={<WikipediaSearch />} />
```
Comment (MaryBruff) [inline src/App/App.js:16]: I really like the inclusion of the article in the url!

**Comment by MaryBruff** [comment]
Sik! Glad you could figure out the pageid for the article url! It is better than what I had, and I like the article's inclusion in the url.

Also, Router is super close to being done! I could be wrong but we may just need a way to clear the state when you go back to search so the card/search is removed.

Otherwise, most excellent job!!

File: src/UserView/UserView.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  return (
+    <div>
+      {!showFavorites ? (
+        <button onClick={handleShowFavorites}>Show me only my favorites</button>
+      ) : (
+        <button onClick={handleShowAllControversies}>Show me all of my controversies again</button>
+      )}
+
+      <div id="resultsList">
+        {showFavorites && controversies.length > 0 ? (
```
Comment (nicktu12) [inline src/UserView/UserView.js:24]: nice conditional rendering happening here!

File: src/Search/Search.js [hunk 3]
```diff
@@ -67,31 +68,28 @@ function WikipediaSearch() {
...
         })
         .catch((error) => {
           console.error('There was a problem with the fetch operation:', error);
         });
-    
+    }
   }, [initialResults]);
 
 // This useEffect will run when the page state is updated to find the controversies
   useEffect(() => {
```
Comment (nicktu12) [inline src/Search/Search.js:80]: This useEffect is pretty complicated - can `words` and `findMatchingSections` be defined outside of the use effect (for readability?) I can picture some of this other functionality could be abstracted into their own methods as well

**Pull request merged**